privaterelay/management/commands/update_phone_remaining_stats.py

```diff
@@ -0,0 +1,59 @@
+from datetime import datetime, timedelta, timezone
+from allauth.socialaccount.models import SocialAccount
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+if settings.PHONES_ENABLED:
+    from phones.models import RelayNumber
+
+from privaterelay.views import update_fxa
+
+
+def update_phone_remaining_stats():
+    if not settings.PHONES_ENABLED:
+        return []
+    accounts_with_phones = []
+    for sub_with_phone in settings.SUBSCRIPTIONS_WITH_PHONE.split(","):
+        social_accounts = SocialAccount.objects.filter(
+            extra_data__icontains=sub_with_phone
+        )
+        for social_account in social_accounts:
+            accounts_with_phones.append(social_account)
+    updated_profiles = []
+    for social_account in accounts_with_phones:
+        profile = social_account.user.profile_set.first()
+        # Has it been more than 30 days since we last checked the user's phone
+        # subscription?
+        if (
+            not profile.date_phone_subscription_checked
+            or profile.date_phone_subscription_checked
+            <= datetime.now(timezone.utc) - timedelta(30)
+        ):
+            # Re-fetch all FXA data into profile so we can see if the user still has
+            # a phone subscription.
+            update_fxa(social_account)
+            profile.date_phone_subscription_checked = datetime.now(timezone.utc)
```

Comment:
I had to think about this, in the context of the test below.

* User has paid for phones, or is otherwise eligible (foxfooder, etc.)
* They haven't setup a Relay phone yet.
* We set that we've checked their profile, don't look again for 30 days.
* 28 days later, they get a Relay number, initialized to the `MAX_x_PER_BILLING_CYCLE`, but there is only 2 days left in the cycle
* 2 days later, this runs again, they are reset to `MAX_x_PER_BILLING_CYCLE`

That seems OK to me.